frontend/src/components/problem/editor/ProblemDisplay.tsx

```diff
@@ -0,0 +1,203 @@
+  '### Example Test Case',
+  '**Input**',
+  '`name`: `value`',
+  '**Output**',
+  '`value`',
+  '**Explanation**',
+  'Replace this line with an explanation of this example test case.',
+].join('\n\n');
+
+const MainContent = styled.div`
+  text-align: left;
+  padding: 20px;
+  flex: 6;
+`;
+
+const TopButtonsContainer = styled.div`
+  margin-left: auto;
+`;
+
+const SettingsContainerHighPadding = styled(SettingsContainer)`
+  padding: 3rem;
+`;
+
+const TitleDescriptionSeparator = styled.hr`
+  margin: 1rem 0;
+  border: none;
+  border-top: 1px solid ${({ theme }) => theme.colors.border};
+`;
+
+const StyledMarkdownEditor = styled(MarkdownEditor)`
+  p {
+    font-family: ${({ theme }) => theme.font};
+  }
+
+  // The specific list of attributes to have dark text color.
+  .ProseMirror > p, blockquote, h1, h2, h3, ul, ol, table {
+    color: ${({ theme }) => theme.colors.text};
+  }
+`;
+
+const TextButtonLink = styled(TextButton)`
+  color: ${({ theme }) => theme.colors.gray};
+  padding: 0;
+  margin-top: 30px;
+  text-decoration: underline;
+  text-align: left;
+`;
+
+const InlineProblemIcon = styled(InlineIcon)`
+  margin: 10px;
+  height: 1rem;
+`;
+
+type ProblemDisplayParams = {
+  problem: Problem,
+  actionText: string,
+  onClick: (newProblem: Problem) => void,
+  editMode: boolean,
+};
+
+function ProblemDisplay(props: ProblemDisplayParams) {
+  const {
+    problem, onClick, actionText, editMode,
+  } = props;
+
+  const history = useHistory();
+  const { firebaseUser } = useAppSelector((state) => state.account);
+  const problemEditable = useProblemEditable(firebaseUser, problem);
+
+  const [newProblem, setNewProblem] = useState<Problem>(problem);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [helpModal, setHelpModal] = useState<boolean>(false);
+
+  // Variable used to force refresh the editor.
+  const [refreshEditor, setRefreshEditor] = useState<number>(0);
+
+  // Handle updating of normal text fields
+  const handleChange = (e: any) => {
+    const { name, value } = e.target;
+    setNewProblem({
+      ...newProblem,
+      [name]: value,
+    });
+  };
+
+  // Handle description change
+  const handleDescriptionChange = (value: string) => handleChange({ target: { name: 'description', value } });
+
+  return (
+    <>
+      <ProblemHelpModal
+        show={helpModal}
+        exitModal={() => setHelpModal(false)}
+      />
+
+      <MainContent>
+        <FlexBareContainer>
+          <SmallHeaderText>Problem</SmallHeaderText>
+          <InlineProblemIcon
+            onClick={() => setHelpModal(true)}
+          >
+            help_outline
+          </InlineProblemIcon>
+          <TopButtonsContainer>
+            <InvertedSmallButton
+              onClick={() => {
+                if (JSON.stringify(problem) === JSON.stringify(newProblem)
+                  || window.confirm('Go back? Your unsaved changes will be lost.')) {
```

Comment:
Add a `// eslint-disable-next-line no-alert` line above this to remove the ESLint warning.